salt/minion.py

```diff
@@ -984,8 +991,14 @@ def _spawn_minions(self):
                                                 loaded_base_name='salt.loader.{0}'.format(s_opts['master']),
                                                 jid_queue=self.jid_queue,
                                                )
-            self.minions.append(minion)
-            self.io_loop.spawn_callback(self._connect_minion, minion)
+
+            self._connect_minion(minion)
+
+        if not self.minions:
+            err = ('Minion unable to successfully connect to '
+                   'a Salt Master.  Exiting.')
+            log.error(err)
+            raise SaltSystemExit(code=42, msg=err)
```

Comment:
We have constants for this. Please do not use hard-coded error codes.

Reply:
This particular return code is used in several other places in the code base, as is, not referenced by a constant, including in minion.py.